apps/minifront/src/components/header/menu/provider.tsx

```diff
@@ -1,29 +1,27 @@
 import { cn } from '@repo/ui/lib/utils';
 import * as NavigationMenu from '@radix-ui/react-navigation-menu';
 import { getChainId } from '../../../fetchers/chain-id';
-import { useCallback, useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 import { itemStyle, triggerStyle, dropdownStyle, linkStyle } from './nav-style';
 import { Link1Icon } from '@radix-ui/react-icons';
-import { getPraxManifest, getPraxOrigin } from '../../../prax';
+import { usePenumbra } from '@penumbra-zone/react';
 
 export const ProviderMenu = () => {
+  const penumbra = usePenumbra();
   const [chainId, setChainId] = useState<string | undefined>();
   const [providerManifest, setProviderManifest] = useState<ProviderManifest>();
-  const [providerOrigin] = useState(getPraxOrigin());
 
   const [manifestIconUnavailable, setManifestIconUnavailable] = useState<boolean>();
 
-  const disconnect = useCallback(() => {
-    console.log('unimplemented');
-    //window[Symbol.for('penumbra')][providerOrigin].disconnect(),
-  }, [providerOrigin]);
-
   useEffect(() => {
-    void getPraxManifest().then(m => setProviderManifest(m as ProviderManifest));
+    if (penumbra?.manifest)
+      void fetch(penumbra.manifest).then(async m =>
+        setProviderManifest((await m.json()) as ProviderManifest),
+      );
     void getChainId().then(setChainId);
-  }, []);
+  }, [penumbra?.manifest]);
 
-  if (!providerManifest) return null;
+  if (!penumbra || !providerManifest) return null;
 
   return (
     <NavigationMenu.Root className='max-w-96'>
```

Comment:
issue (minor): something in this component triggers the html warning in the devtools: `validateDOMNesting(...): <li> cannot appear as a descendant of <li>`. Small but annoying issue